modules/openy_features/openy_block/modules/openy_block_date/src/EntityOperations.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+namespace Drupal\openy_block_date;
+
+use Drupal\Component\Datetime\TimeInterface;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\DependencyInjection\ContainerInjectionInterface;
+use Drupal\Core\Entity\Display\EntityViewDisplayInterface;
+use Drupal\Core\Entity\EntityInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\Core\Datetime\DrupalDateTime;
+
+/**
+ * Defines a class for reacting to entity events.
+ */
+class EntityOperations implements ContainerInjectionInterface {
+
+  /**
+   * Date block tag.
+   */
+  const TAG = 'block_date';
+
+  /**
+   * Time.
+   *
+   * @var \Drupal\Component\Datetime\TimeInterface
+   */
+  protected $time;
+
+  /**
+   * Config factory.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  protected $configFactory;
+
+  /**
+   * EntityOperations constructor.
+   *
+   * @param \Drupal\Component\Datetime\TimeInterface $time
+   *   Time.
+   * @param \Drupal\Core\Config\ConfigFactoryInterface
+   *   Config factory.
+   */
+  public function __construct(TimeInterface $time, ConfigFactoryInterface $configFactory) {
+    $this->time = $time;
+    $this->configFactory = $configFactory;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('datetime.time'),
+      $container->get('config.factory')
+    );
+  }
+
+  /**
+   * Acts on hook_ENTITY_TYPE_view_alter().
+   *
+   * @param array $build
+   *   Build.
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   Entity.
+   * @param \Drupal\Core\Entity\Display\EntityViewDisplayInterface $display
+   *   Display.
+   *
+   * @return array
+   *   Build.
+   */
+  public function viewAlter(array $build, EntityInterface $entity, EntityViewDisplayInterface $display) {
+    $requestTime = $this->time->getRequestTime();
+
+    $dates = [];
+    $fields_date = ['field_start_date', 'field_end_date'];
+    foreach ($fields_date as $field) {
+      // Set tags with timestamps of future block changes.
+      $dateTime = DrupalDateTime::createFromFormat(DATETIME_DATETIME_STORAGE_FORMAT, $entity->$field->value, 'UTC');
```

Comment:
Why hardcoded UTC?
It doesn't work now when I'm posting from Ukraine.

Reply:
No hardcode.
`DATETIME_DATETIME_STORAGE_FORMAT` is stored in UTC by Drupal design.

Reply:
ok, tested. It works